src/infrasys/h5_time_series_storage.py

```diff
@@ -115,28 +124,129 @@ def add_time_series(
 
         Parameters
         ----------
-        metadata : TimeSeriesMetadata
+        metadata : :class:`~infrasys.time_series_models.TimeSeriesMetadata`
             Metadata for the time series
-        time_series : TimeSeriesData
+        time_series : :class:`~infrasys.time_series_models.TimeSeriesData`
             Time series data to store
         context : Any, optional
-            Optional context parameter
+            Optional context parameter, by default None
+
+        See Also
+        --------
+        :meth:`_add_time_series_dispatch` : Dispatches the call to the correct handler based on metadata type.
+        """
+        if context is not None:
+            self._add_time_series_dispatch(metadata, time_series, context=context)
+        else:
+            with self.open_time_series_store() as file_handle:
+                self._add_time_series_dispatch(metadata, time_series, context=file_handle)
+
+    @singledispatchmethod
+    def _add_time_series_dispatch(
+        self,
+        metadata: TimeSeriesMetadata,
+        time_series: TimeSeriesData,
+        context: Any = None,
+    ) -> None:
+        """Dispatches the call to the correct handler based on metadata type.
+
+        Parameters
+        ----------
+        metadata : :class:`~infrasys.time_series_models.TimeSeriesMetadata`
```

Comment:
Why is the `:class: ...` needed? I thought that we are getting hyperlinks based on type annotations.

Reply:
I think it only catches up standard library types, but I could be wrong.

Reply:
If the type is included in the docs, it will work.
https://nrel.github.io/infrasys/reference/api/system.html#infrasys.system.System.add_component

It does seem that we need to specify the type in the `Parameters` section for maximum clarity. This example does not do that:
https://nrel.github.io/infrasys/reference/api/system.html#infrasys.system.System.add_supplemental_attribute